Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraintPalmUp.cs

```diff
@@ -163,5 +197,125 @@ protected override bool IsValidController(IMixedRealityController controller)
 
             return true;
         }
+
+        /// <summary>
+        /// Checks to see if the palm is currently facing the user
+        /// </summary>
+        /// <param name="jointedHand"></param>
+        /// <param name="palmPose"></param>
+        /// <param name="palmCameraAngle"></param>
+        /// <returns></returns>
+        private bool IsPalmFacingUpwards(IMixedRealityHand jointedHand, MixedRealityPose palmPose, float palmCameraAngle)
+        {
+            if (requireFlatHand)
+            {
+                // Check if the triangle's normal formed from the palm, to index, to ring finger tip roughly matches the palm normal.
+                MixedRealityPose indexTipPose, ringTipPose;
+
+                if (jointedHand.TryGetJoint(TrackedHandJoint.IndexTip, out indexTipPose) &&
+                    jointedHand.TryGetJoint(TrackedHandJoint.RingTip, out ringTipPose))
+                {
+                    var handNormal = Vector3.Cross(indexTipPose.Position - palmPose.Position,
+                                                   ringTipPose.Position - indexTipPose.Position).normalized;
+                    handNormal *= (jointedHand.ControllerHandedness == Handedness.Right) ? 1.0f : -1.0f;
+
+                    if (Vector3.Angle(palmPose.Up, handNormal) > flatHandThreshold)
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            // Check if the palm angle meets the prescribed threshold
+            return palmCameraAngle < facingCameraTrackingThreshold;
+        }
+
+        /// <summary>
+        /// Checks to see if the user is currently gazing at the activation point; it first attempts to do so 
+        /// using eyegaze, and then falls back to head-based gaze if eyegaze isn't available for use.
+        /// </summary>
+        /// <param name="jointedHand"></param>
+        /// <returns></returns>
+        private bool IsUserGazingAtActivationPoint(IMixedRealityHand jointedHand)
+        {
+            Ray gazeRay;
+
+            if (InputRayUtils.TryGetRay(InputSourceType.Eyes, Handedness.Any, out gazeRay) || InputRayUtils.TryGetRay(InputSourceType.Head, Handedness.Any, out gazeRay))
+            {
+                // Generate the hand plane that we're using to generate a distance value.
+                // This is done by using the index knuckle, pinky knuckle, and wrist
+                MixedRealityPose indexKnuckle;
```

Comment:
You should probably be using the hand bounds service to generate the plane.

Reply:
Are the thoughts there that the plane would be generated based on the bounds center + two other generated points that refer to the min/max vector values to get a "Center" plane? If not, what are your thoughts here?